python/ray/serve/backend_worker.py

```diff
@@ -251,6 +253,31 @@ def get_runner_method(self, request_item: Query) -> Callable:
             return self.callable
         return getattr(self.callable, method_name)
 
+    async def ensure_serializable_response(self, response: Any) -> Any:
+        if isinstance(response, starlette.responses.StreamingResponse):
+            # response contains a generator/iterator which is not serializable.
+            # Exhaust the generator/iterator and store the results in a buffer.
+            body_buffer = []
+
+            async def mock_send(message):
+                if (message["type"] == "http.response.body"):
+                    body_buffer.append(message["body"])
+
+            async def mock_receive():
+                # This is called in a while loop in response(), so sleep a bit.
+                await asyncio.sleep(1)
```

Comment:
we definitely don't need the sleep here. can we remove it? or would it error?

Reply:
I used to not have the sleep and it doesn't work unfortunately, on my laptop it hanged and my fans got loud.  Starlette calls it in a tight loop here https://github.com/encode/starlette/blob/master/starlette/responses.py#L204 (just to check for http disconnects) alongside the actual sending process, and finishes when the sending is done.  Ideally I could just await forever here, what would you suggest?

Reply:
ah i see. i think case let's just set a event that will never fire and add comments to explain the reasoning behind it.

Reply:
Nice, that sounds like the correct way to do it.  I've added this.
```python
            async def mock_receive():
                # This is called in a tight loop in response() just to check
                # for an http disconnect.  So rather than return immediately
                # we should suspend execution to avoid wasting CPU cycles.
                never_set_event = asyncio.Event()
                await never_set_event.wait()
```